src/ingest-pipeline/misc/tools/split_and_create.py

```diff
@@ -199,8 +200,38 @@ def populate(row, source_entity, entity_factory, dryrun=False):
         print(row_df)
     else:
         kid_path = Path(entity_factory.get_full_path(uuid))
+
     row_df.to_csv(kid_path / f"{uuid}-metadata.tsv", header=True, sep="\t", index=False)
     extras_path = kid_path / "extras"
+
+    # ToDo read all component metadata files find the matching row for old_data_path and then refactor info
+    #  as main metadata above before creating the new component metadata tsv
+    if components is not None:
+        for component in components:
+            component_df = pd.read_csv(component.get("metadata-file"))
+            row_component = component_df.query(f'data_path=="{old_data_path}"')
+            assert (
+                len(row_component) == 0
```

Comment:
I see your TODO at the top of this function so maybe this is not completely ready yet, but I'm a little confused here. On line 211 you read in the component metadata file and then on line 212 you query it for a `data_path` field matching the `old_data_path` found in the primary TSV. I would be expecting you to want this to find a match (indicating that the component TSV has a row matching the row we're looking at in the parent TSV) and for this assert to read `len(row_components) == 1` rather than 0. Am I just confused?

Reply:
You are right, it should be 1, somehow I had it backwards. Removing the TODO.